### PR TITLE
Prevent setting reversed domain on Scale.Time

### PR DESCRIPTION
**Pull request by jtlan**

### CODE REVIEW

File: src/scales/timeScale.ts [hunk 1]
```diff
@@ -31,6 +31,9 @@ export module Scale {
     protected _setDomain(values: any[]) {
       // attempt to parse dates
       values = values.map(this._typeCoercer);
+      if (values[1] < values[0]) {
+        throw new Error("Scale.Time domain values must be in chronological order");
+      }
```
Comment (bluong) [inline src/scales/timeScale.ts]: This is because I assume we don't have reversable timeScales yet?

File: test/scales/timeScaleTests.ts [hunk 2]
```diff
@@ -3,62 +3,67 @@
...
+    ticks = scale.tickInterval(d3.time.minute);
+    assert.equal(ticks.length, 61, "generated correct number of ticks");
+    ticks = scale.tickInterval(d3.time.minute, 10);
+    assert.equal(ticks.length, 7, "generated correct number of ticks");
+    // 1 minute span
+    scale.domain([new Date(2000, 0, 1, 0, 0, 0, 0), new Date(2000, 0, 1, 0, 1, 0, 0)]);
+    ticks = scale.tickInterval(d3.time.second);
+    assert.equal(ticks.length, 61, "generated correct number of ticks");
+  });
 });
```
Comment (bluong) [inline test/scales/timeScaleTests.ts]: Why is there so much revamp here?

Comment (bluong) [inline test/scales/timeScaleTests.ts]: Whoa never mind.  Just a bunch of formatting fixes.

File: test/tests.js [hunk 3]
```diff
@@ -6102,6 +6102,10 @@ describe("TimeScale tests", function () {
         checkDomain(["October 1, 2014", "November 1, 2014"]);
         checkDomain(["Oct 1, 2014", "Nov 1, 2014"]);
     });
+    it("can't set reversed domain", function () {
+        var scale = new Plottable.Scale.Time();
+        assert.throws(function () { return scale.domain(["1985-10-26", "1955-11-05"]); }, "chronological");
```
Comment (bluong) [inline test/tests.js]: Can we use the () => notation?  Just cleaner

**Comment by bluong** [comment]
Minor

File: test/tests.js (hunk 3 above)
Comment (jtlan) [inline test/tests.js]: this is a JS file.

File: src/scales/timeScale.ts (hunk 1 above)
Comment (jtlan) [inline src/scales/timeScale.ts]: yes.

File: test/tests.js (hunk 3 above)
Comment (bluong) [inline test/tests.js]: Whoops.

**Comment by crmorford** [comment]
gud

**Pull request merged**